pkg/apis/planetscale/v2/vitessshard_types.go

```diff
@@ -308,24 +308,50 @@ type VitessShardStatus struct {
 	// HasMaster is a condition indicating whether the Vitess topology
 	// reflects a master for this shard.
 	HasMaster corev1.ConditionStatus `json:"hasMaster,omitempty"`
+
 	// HasInitialBackup is a condition indicating whether the initial backup
 	// has been seeded for the shard.
 	HasInitialBackup corev1.ConditionStatus `json:"hasInitialBackup,omitempty"`
+
 	// MasterAlias is the tablet alias of the master according to the global
 	// shard record. This could be empty either because there is no master,
 	// or because the shard record could not be read. Check the HasMaster
 	// condition whenever the distinction is important.
 	MasterAlias string `json:"masterAlias,omitempty"`
 
+	// BackupLocations reports information about the backups for this shard in
+	// each backup location.
+	BackupLocations []*ShardBackupLocationStatus `json:"backupLocations,omitempty"`
+
 	// Idle is a condition indicating whether the shard can be turned down.
 	// If Idle is True, the shard is not part of the active shard set
 	// (partitioning) for any tablet type in any cell, so it should be safe
 	// to turn down the shard.
 	Idle corev1.ConditionStatus `json:"idle,omitempty"`
 
-	// BackupLocations reports information about the backups for this shard in
-	// each backup location.
-	BackupLocations []*ShardBackupLocationStatus `json:"backupLocations,omitempty"`
+	// Conditions is a map of all VitessShard specific conditions we want to set and monitor.
+	Conditions map[VitessShardConditionType]*VitessShardCondition `json:"conditions,omitempty"`
+}
+
+// VitessShardConditionType is a valid value for VitessShardCondition.Type
+type VitessShardConditionType string
+
+// VitessShardCondition contains details for the current condition of this VitessShard.
+type VitessShardCondition struct {
+	// Type is the type of the condition.
+	Type VitessShardConditionType `json:"type"`
```

Comment:
This is mostly redundant since we decided to use a map. I know it's just following the convention, but we already broke with convention by using a map, so I think we should only include this if it makes sense from first principles in a map-based design.

Do you think it's generally worthwhile to duplicate the map key inside the structured value of an entry? In my experience, there are pros and cons of both options.

Reply:
I think if we don't care about conventions at all, then it doesn't provide any extra benefit. In fact, having it seems to encourage mutability, and it would be very odd if the type of a condition mutated but its key stays the same. So I would be in favor of removing it if you are as well. It seems error-prone. WDYT?

Reply:
OK let's remove then.

Reply:
Removed https://github.com/planetscale/vitess-operator/pull/31/commits/ef506e72fe8f258c9628fd1d9d2c9d15c5f12acc.